Scripts/Process/MomentProcessor.cs

```diff
@@ -0,0 +1,156 @@
+            /// Executes the processes in the Update game logic part of the Unity game loop.
+            /// </summary>
+            OnUpdate,
+            /// <summary>
+            /// Executes the processes in the LateUpdate game logic part of the Unity game loop.
+            /// </summary>
+            OnLateUpdate,
+            /// <summary>
+            /// Executes the processes in the camera PreCull scene rendering part of the Unity game loop.
+            /// </summary>
+            OnPreCull,
+            /// <summary>
+            /// Executes the processes in the camera PreRender scene rendering part of the Unity game loop.
+            /// </summary>
+            OnPreRender
+        }
+
+        [Tooltip("The moment in which to process the given processes.")]
+        public MomentToProcess momentToProcess = MomentToProcess.OnPreRender;
+        [Tooltip("A collection of Moments to process.")]
+        public MomentProcess[] processes = new MomentProcess[0];
+
+        protected MomentToProcess subscribedMoment;
+
+        protected virtual void OnEnable()
+        {
+            subscribedMoment = MomentToProcess.None;
+            ManageSubscriptions();
+        }
+
+        protected virtual void OnDisable()
+        {
+            UnsubscribeMoment();
+        }
+
+        protected virtual void FixedUpdate()
+        {
+            if (momentToProcess == MomentToProcess.OnFixedUpdate)
+            {
+                Process();
+            }
+        }
+
+        protected virtual void Update()
+        {
+            if (momentToProcess == MomentToProcess.OnUpdate)
+            {
+                Process();
+            }
+        }
+
+        protected virtual void LateUpdate()
+        {
+            if (momentToProcess == MomentToProcess.OnLateUpdate)
+            {
+                Process();
+            }
+            ManageSubscriptions();
+        }
+
+        protected virtual void OnCameraPreRender(Camera givenCamera)
+        {
+            Process();
+        }
+
+        protected virtual void OnCameraPreCull(Camera givenCamera)
+        {
+            Process();
+        }
+
+        /// <summary>
+        /// The ManageSubscriptions method handles subscribing and unsubscribing to the relevant camera events.
+        /// </summary>
+        protected virtual void ManageSubscriptions()
+        {
+            if (subscribedMoment != momentToProcess)
+            {
+                UnsubscribeMoment();
+                SubscribeMoment();
+            }
+        }
+
+        /// <summary>
+        /// The UnsubscribeMoment method handles unsubscribing to the chosen subscribed moment event.
+        /// </summary>
+        protected virtual void UnsubscribeMoment()
+        {
+            switch (subscribedMoment)
+            {
+                case MomentToProcess.OnPreRender:
+                    Camera.onPreRender -= OnCameraPreRender;
+                    subscribedMoment = MomentToProcess.None;

```

Comment:
It looks like this line is in both cases but should be done regardless of the current value of `subscribedMoment`. I think we should move it to be done after this `switch`.

Reply:
👍